File: jambeth/jambeth-log/src/main/java/com/koch/ambeth/log/ILoggerHistory.java
package com.koch.ambeth.log;

public interface ILoggerHistory
{
	boolean addLogHistory(ILogger logger, Object contextHandle, String logTextForHistory);

	boolean debugOnce(ILogger log, Object contextHandle, String logTextForHistory);

	boolean infoOnce(ILogger log, Object contextHandle, String logTextForHistory);

	boolean warnOnce(ILogger log, Object contextHandle, String logTextForHistory);

	boolean errorOnce(ILogger log, Object contextHandle, String logTextForHistory);
}

File: ambeth/Ambeth.Log/ambeth/log/ILoggerHistory.cs
using System;

namespace De.Osthus.Ambeth.Log
{
    public interface ILoggerHistory
    {
        bool AddLogHistory(ILogger logger, Object contextHandle, String logTextForHistory);

        bool DebugOnce(ILogger log, Object contextHandle, String logTextForHistory);

        bool InfoOnce(ILogger log, Object contextHandle, String logTextForHistory);

        bool WarnOnce(ILogger log, Object contextHandle, String logTextForHistory);

        bool ErrorOnce(ILogger log, Object contextHandle, String logTextForHistory);
    }
}